### ISSUE TITLE
Missing "-lm" linking

### ISSUE THREAD

**Issue by TarekTaha** [queezythegreat/arduino-cmake#2]
I kept on gettting:: In function `__lesf2':
(.text.fplib+0x0): multiple definition of`__ltsf2'

until I linked the "-lm" to the code 

list(APPEND ALL_LIBS "-lm" ${CORE_LIB} ${INPUT_LIBS})

I love this CMAKE scripts for Arduino, very smart, I can now use my favorite IDE "Qtcreator" to compile arduino code

Thanks for your effort

**Comment by queezythegreat** [comment]
Hi,

Nice to hear that my build system is useful to people :) As far as that library, you should be able to do the following:

set(${TARGET_NAME}_LIBS m)

This solution is more generic.. (although I haven't tested it).

**Comment by TarekTaha** [comment]
I already tried that, but "m" library should be linked at the end as the last library otherwise the build will fail.

**Comment by queezythegreat** [comment]
Unfortunately this more related to the ld linker than CMake. ld requires a strict order when linking static libraries. I've had this problem many times in big projects, that have a lot of static libraries.

CMake is smart (creates the correct order) when linking static libraries, but only when it knows about the dependencies. There is a way to create a dependency chain between static libraries but I'm not sure if there is a good way to do this generically in this case.

It would be nice to know more about your specific situation. Could you describe what your building and give me you're CMake configuration file. 

I'm going to think about this, and try to find a generic solution which will solve the problem you've encountered.

**Comment by treaves** [comment]
Any updates on this?  I now am running into this.  You can see my project at: 
https://code.launchpad.net/~treaves/+junk/bt_cmake

**Comment by queezythegreat** [comment]
I've got a hackish solution to the problem :D If you add the following, before any definition (in the main CMakeLists.txt before all add_subdirectory):

``` cmake
link_libraries(m)
```

Then the library **m** will be the last library linked. 

Here is the diff:

``` diff
=== modified file 'CMakeLists.txt'
--- CMakeLists.txt  2012-01-20 21:54:33 +0000
+++ CMakeLists.txt  2012-01-22 17:30:45 +0000
@@ -41,6 +41,8 @@
    ${CMAKE_SOURCE_DIR}/libraries/PID_Beta6
 )

+link_libraries(m)
+
 add_subdirectory(libraries)
 add_subdirectory(src)


```

By the way I could not get your project to compile, but I was able to create a build system (by changing the required Arduino version and board ID). The linker flags in the build system had **-lm** as the last linker option, so it should fix your problem.

I wonder how the Arduino IDE solves this problem, maybe I should add the **m** library as a default dependency for the arduino core library.
